### PR TITLE
Implement Active Set Changes RPC Method

### PR DESCRIPTION
**Pull request by rauljordan**
Part of #3045 

---

# Description

**Write why you are making the changes in this pull request**

This PR uses the archived data to fetch validator active set changes in an RPC request accordingly

### CODE REVIEW

File: beacon-chain/archiver/service.go [hunk 1]
```diff
@@ -101,60 +100,18 @@ func (s *Service) archiveCommitteeInfo(ctx context.Context, headState *pb.Beacon
 
 // We archive active validator set changes that happened during the epoch.
 func (s *Service) archiveActiveSetChanges(ctx context.Context, headState *pb.BeaconState) error {
```
Comment (nisdas) [inline beacon-chain/archiver/service.go]: how often are you going to be calling this function ? 
this loops through the validator set 3 times now , which 
might be a problem if this is called frequently

File: beacon-chain/core/validators/validator.go [hunk 2]
```diff
@@ -128,3 +128,71 @@ func SlashValidator(state *pb.BeaconState, slashedIdx uint64, whistleBlowerIdx u
...
+		maxWithdrawableEpoch := mathutil.Max(val.WithdrawableEpoch, currentEpoch+params.BeaconConfig().EpochsPerSlashingsVector)
+		if val.WithdrawableEpoch == maxWithdrawableEpoch && val.Slashed {
+			slashed = append(slashed, uint64(i))
+		}
+	}
+	return slashed
+}
+
+// ExitedValidatorIndices determines the indices exited during the current epoch.
+func ExitedValidatorIndices(state *pb.BeaconState) ([]uint64, error) {
```
Comment (nisdas) [inline beacon-chain/core/validators/validator.go]: you loop through the validator set 3 times here, anyway we can do it all in a single iteration ?

File: beacon-chain/archiver/service.go (hunk 1 above)
Comment (rauljordan) [reply beacon-chain/archiver/service.go]: Only once per epoch, and it's O(3N) = O(N), so it should be alright

File: beacon-chain/core/validators/validator.go (hunk 2 above)
Comment (rauljordan) [reply beacon-chain/core/validators/validator.go]: Will try to consolidate

Comment (rauljordan) [reply beacon-chain/core/validators/validator.go]: I can't find a way to break it up - it seems like each next iteration depends on the previous one to complete

**Pull request merged**